runtime/gc_glue_java/ConcurrentMarkingDelegate.cpp

```diff
@@ -450,4 +450,31 @@ MM_ConcurrentMarkingDelegate::concurrentClassMark(MM_EnvironmentBase *env, bool
 }
 #endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
 
+void *
+MM_ConcurrentMarkingDelegate::sealAllocCache(MM_EnvironmentBase *env, MM_TLHAllocationSupport *tlh)
+{
+	J9JavaVM *javaVM = (J9JavaVM *)env->getLanguageVM();
+
+	uint8_t *lastCacheObj = (uint8_t *) tlh->getAlloc();
+	tlh->setAlloc((void *)((uintptr_t) lastCacheObj + OMR_MINIMUM_OBJECT_SIZE));
+
+	J9Class *clazz = J9VMJAVALANGOBJECT_OR_NULL(javaVM);
+	uintptr_t const headerSize = J9JAVAVM_OBJECT_HEADER_SIZE(javaVM);
+	j9object_t instance = (j9object_t) lastCacheObj;
+
+	memset(instance, 0, headerSize);
```

Comment:
This zeros mixed object header (and it happen to be class slot only about to be set). Lock word however is not included to the object header but it is object slot. Should zeroing be done for OMR_MINIMUM_OBJECT_SIZE bytes to erase entire object?
Also is there an assumption that java/lang/object size is smaller or equal OMR_MINIMUM_OBJECT_SIZE (guaranteed pre-reserved)? We can add assertion somewhere (in the path to be called once, not here) to verify this fact.

Reply:
Good catch for clearing.
But I would not be too concerned about size violations. If there is a such a problem, VM would be in trouble at many other places, not just SATB premarking related.

Reply:
There's actually already an assert I put in place for this. When we return from this call, we have:

```C++
void* lastTLHobj = _concurrentDelegate.sealAllocCache(env, tlh);
uintptr_t lastTLHobjSize = _extensions->objectModel.getConsumedSizeInBytesWithHeader((omrobjectptr_t)lastTLHobj);
Assert_MM_true(OMR_MINIMUM_OBJECT_SIZE == lastTLHobjSize);
```
 Hard to know since changes are split with OMR.

Reply:
Clearing changed to OMR_MINIMUM_OBJECT_SIZE as suggested